packages/core/src/codewhispererChat/tools/executeBash.ts

```diff
@@ -307,3 +365,17 @@ export class ExecuteBash {
         updates.end()
     }
 }
+
+class AtomicBoolean {
```

Comment:
? javascript is single-thread, what is the purpose of this?

Reply:
yes, javascript is single-thread, but we have the use case that one process can send bash command execution output to stdout and stderr, which the streaming of stdout and stderr is async.

Reply:
javascript has a lot of asynchrony, but each async execution context is executed fully until a `async` call, and is not "preemptible". `getAndSet()` is not doing anything useful here.

Reply:
see https://github.com/aws/aws-toolkit-vscode/pull/6995/files#r2037764657